packages/components/src/components/assets/CoinLogo/CoinLogo.tsx

```diff
@@ -24,15 +25,44 @@ const SvgWrapper = styled.div<Omit<CoinLogoProps, 'symbol'>>`
     }
 `;
 
-export const CoinLogo = ({ symbol, className, size = 32, ...rest }: CoinLogoProps) => (
-    <SvgWrapper className={className} size={size} {...rest}>
-        <ReactSVG
-            src={COINS[symbol]}
-            beforeInjection={svg => {
-                svg.setAttribute('width', `${size}px`);
-                svg.setAttribute('height', `${size}px`);
-            }}
-            loading={() => <span className="loading" />}
-        />
-    </SvgWrapper>
+const StyledIconImage = styled.img<CoinLogoProps>(
+    ({ size }) => `
+        width: ${size}px;
+        height: ${size}px;
+    `,
 );
+
+export const CoinLogo = ({ symbol, iconUrl, className, size = 32, ...rest }: CoinLogoProps) => {
+    const [isLoading, setIsLoading] = useState(true);
+
+    if (symbol) {
+        return (
+            <Wrapper className={className} size={size} {...rest}>
+                <ReactSVG
+                    src={COINS[symbol]}
+                    beforeInjection={svg => {
+                        svg.setAttribute('width', `${size}px`);
+                        svg.setAttribute('height', `${size}px`);
+                    }}
+                    loading={() => <span className="loading" />}
```

Comment:
Is there a reason why to use classname and not normal styled component?

Reply:
I don't know the author's intentions, you are referring to the code that isn't mine. However it didn't look safe to just remove it.

Reply:
OK, makes sense